scikeras/_utils.py

```diff
@@ -155,19 +156,86 @@ def get_metric_full_name(name: str) -> str:
     # deserialize returns the actual function, then get it's name
     # to keep a single consistent name for the metric
     if name == "loss":
-        # may be passed "loss" from thre training history
+        # may be passed "loss" from training history
         return name
     return getattr(deserialize_metric(name), "__name__")
 
 
-def _get_default_args(func):
-    signature = inspect.signature(func)
-    return {
-        k: v.default
-        for k, v in signature.parameters.items()
-        if v.default is not inspect.Parameter.empty
-    }
+def _windows_upcast_ints(
+    arr: Union[List[np.ndarray], np.ndarray]
+) -> Union[List[np.ndarray], np.ndarray]:
+    # see tensorflow/probability#886
+    def _upcast(x):
+        return x.astype("int64") if x.dtype == np.int32 else x
+
+    if isinstance(arr, np.ndarray):
+        return _upcast(arr)
+    else:
+        return [_upcast(x_) for x_ in arr]
+
 
+def route_params(
+    params: Dict[str, Any],
+    destination: str,
+    pass_filter: Union[Iterable[str], None] = None,
+) -> Dict[str, Any]:
+    """Route and trim parameter names.
+
+    Parameters
+    ----------
+    params : Dict[str, Any]
+        Parameters to route/filter.
+    destination : str
+        Destination to route to, ex: `build` or `compile`.
+    pass_filter: Union[Iterable[str], None], default None
+        If None, all non-routing `params` are passed. If an iterable,
+        only keys from `params` that are in the iterable are passed.
+        This does not affect routed parameters.
 
-def _windows_upcast_ints(x: np.ndarray) -> np.ndarray:
-    return x.astype("int64") if x.dtype == np.int32 else x
+    Returns
+    -------
+    Dict[str, Any]
+        Filtered parameters, with any routing prefixes removed.
+    """
+    res = dict()
+    for key, val in params.items():
+        if "__" in key:
+            # routed param
+            if key.startswith(destination):
+                new_key = key.replace(destination.strip("__") + "__", "")
```

Comment:
What if the user specifies `model__model__foo` as a parameter? I think I'd like to see

``` python
new_key = key[len(destination + "__"):]
```

Reply:
Yes, this is a good catch. We should change it and test it.